version: '3.8'

services:

  postgresql:
    image: postgres
    container_name: pixlon-postgresql
    restart: 'on-failure'
    ports:
      - "5432:5432"
    environment:
      - POSTGRES_USER=root
      - POSTGRES_PASSWORD=bhulgaya
      - POSTGRES_DB=pixlon
    volumes:
      - postgres-data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U root -d pixlon"]
      interval: 5s
      timeout: 5s
      retries: 5

  app:
    build: .
    container_name: pixlon-app
    restart: 'on-failure'
    ports:
      - "3000:3000"
    environment:
      - DATABASE_URL=postgresql://root:bhulgaya@postgresql:5432/pixlon
      - NEXTAUTH_URL=http://localhost:3000
      - IMAGE_UPLOAD_DIR=/app/public/uploads
    depends_on:
      postgresql:
        condition: service_healthy
    volumes:
      - ./:/app
      

volumes:
  postgres-data:
